### PR TITLE
Reverse a String in Swift

### PR DESCRIPTION
**Pull request by martyav**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language](https://therenegadecoder.com/code/sample-programs-in-every-language/) repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README

## Notes

Feel free to put whatever you want here.

Note that Foundation is imported simply so we can exit early...

### CODE REVIEW

File: archive/s/swift/reverse-string.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+guard CommandLine.arguments.count > 0 else {
+    exit(0)
+}
+
+let usersString = CommandLine.arguments[0]
```
Comment (jrg94) [inline archive/s/swift/reverse-string.swift]: Just tried testing this in the [online Swift playground](https://iswift.org/playground). It seems `CommandLine.arguments[0]` points to the file name (granted I can't seem to pass args to the script anyway). Is this an issue?

File: archive/s/swift/reverse-string.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+let reversedCollection = usersString.reversed()
```
Comment (jrg94) [inline archive/s/swift/reverse-string.swift]: In other [online editors](https://www.jdoodle.com/execute-swift-online), I'm finding that they can't find `reversed()`.

```
jdoodle.swift:8:26: error: value of type 'String' has no member 'reversed'
let reversedCollection = usersString.reversed()
```

**Comment by jrg94** [comment]
Forget what I said! I found an [online editor](https://www.onlinegdb.com/online_swift_compiler#) that runs everything properly (and even lets me use the command line).

**Comment by martyav** [comment]
Re CommandLine picking up the file name -- I had a similar issue when running Swift for Windows. It seems like a very odd bug

File: archive/s/swift/reverse-string.swift (hunk 2 above)
Comment (martyav) [reply archive/s/swift/reverse-string.swift]: I think that's due to them running an earlier version of Swift. Swift4 changed the name of the method from reverse to reversed. Similarly, earlier versions did not have access to a count on String -- you have to call count on the string's characters property

File: archive/s/swift/reverse-string.swift (hunk 1 above)
Comment (martyav) [reply archive/s/swift/reverse-string.swift]: Note that Swift's Strings use Unicode, so they have inherent support for emoji. 

Also, Swift isn't as commonly used on the command line right now, which might be why these online editors have such a poor track record with handling command line arguments

Comment (jrg94) [reply archive/s/swift/reverse-string.swift]: Makes sense! I anticipated running into these kinds of issues with all these different languages.

File: archive/s/swift/reverse-string.swift (hunk 2 above)
Comment (jrg94) [reply archive/s/swift/reverse-string.swift]: Good to know! I'll make a note of that in the article.

**Pull request merged**